### DISCUSSION TITLE
WinUI3 Desktop - How do I create an extensible desktop application?

### DISCUSSION THREAD

**Discussion by williamfigtree** [microsoft/WindowsAppSDK#1575] [Q&A] [answered]
I am trying to build an extensible WinUI3 Desktop application with plugin architecture. The plugins must include their own XAML controls and executable code. The host application should not be dependent on the plugins. Is there a supported solution for this design? At the moment there do not appear to be any templates or tutorials covering this topic for WinUI3.

I have tried to load plugin assemblies at runtime using a custom AssemblyLoadContext. Loading the assembly appears to work however an exception is thrown while instantiating the plugin user control 'PluginRoot'.

```
Exception thrown at 0x00007FF8B02BA839 (KernelBase.dll) in WinUIPluginApp.exe: WinRT originate error - 0x80004005 : 'Cannot locate resource from 'ms-appx:///Plugin/PluginRoot.xaml'.'.
onecore\com\combase\winrt\error\restrictederror.cpp(1014)\combase.dll!00007FF8B1A8EB30: (caller: 00007FF8B1A7FA6C) ReturnHr(1) tid(11614) 8007007E The specified module could not be found.
Exception thrown: 'Microsoft.UI.Xaml.Markup.XamlParseException' in WinRT.Runtime.dll
WinRT information: Cannot locate resource from 'ms-appx:///Plugin/PluginRoot.xaml'.
XAML parsing failed.
```

Ideally the control would be correctly instantiated and put into the visual tree. A replicating repo of this attempt is available [here](https://github.com/williamfigtree/WinUIPluginApp).

**Comment by Daniellled** [comment]
I am having the same issue. Have you resolved this?

**Reply by williamfigtree** [reply]
@Daniellled No I haven't. There is an open issue for the XAML exception here https://github.com/microsoft/microsoft-ui-xaml/issues/6299.

**Comment by BenJKuhn** [comment, marked as answer]
This is now being discussed in the XAML repo in issue https://github.com/microsoft/microsoft-ui-xaml/issues/6299 as noted above, so I'm going to make this one as answered in the sense that the discussion here is complete.
